watershed_workflow/split_hucs.py

```diff
@@ -174,7 +175,18 @@ def polygon(self, i):
                 segs.append(self.segments[s])
 
         ml = shapely.ops.linemerge(segs)
-        assert (type(ml) is shapely.geometry.LineString)
+
+        if type(ml) is shapely.geometry.MultiLineString:
+            logging.debug("the lines may not be contiguous, try to make a new simple\
+                line using the line coordinates.")
+            # Put the sub-line coordinates into a list of sublists
+            coords = [list(line.coords) for line in ml]
+            # Flatten the list of sublists and use it to make a new line
+            newline = shapely.geometry.LineString([i for sublist in coords for i in sublist])
+            assert type(newline) is shapely.geometry.LineString
```

Comment:
this assertion isn't needed -- you just made it a LineString, so no need to check.